src/libraries/System.Text.Json/Common/JsonNumberHandling.cs

```diff
@@ -39,6 +42,9 @@ enum JsonNumberHandling
         /// will be written as their corresponding JSON string representations.
         /// Strings that have escaped characters will be unescaped before reading.
         /// Leading or trailing trivia within the string token, including whitespace, is not allowed.
+        /// <remarks>
+        /// You can do this, but you need to be aware that this will create potentially non-interoperable JSON.
```

Comment:
I personally find this message confusing. I'd suggest something like:

```suggestion
        /// Mentioned values are not defined by the specification. Using this setting may potentially produce JSON which may be non-parsable in other JSON implementations.
```

Reply:
I actually find that better too :)